src/main/java/com/github/davidcarboni/thetrain/helpers/UnionInputStream.java

```diff
@@ -152,6 +152,7 @@ public int read(byte b[], int off, int len) throws IOException {
     public void close() throws IOException {
         logBuilder().info("calling close on union input stream");
         IOUtils.closeQuietly(a);
-        IOUtils.closeQuietly(b);
+        // TODO explain why this is not closed.
```

Comment:
Is there a reason why this is just a TODO right now? Do we know why yet?

Reply:
Yes - Its a flawed approach. The comment in the PR is the how/why - I will i update the code with another commit later to provide an appropriate comment explaining why we do this.